### PR TITLE
BUG: compute Probit loglike, score and hessian in log space

### LINKED ISSUE #10228: BUG: Probit loglike is clipped flat and its Hessian is nan for extreme linear predictors
#### Describe the bug

`Probit` computes its log-likelihood as `log(clip(cdf(z), FLOAT_EPS, 1))`, so every observation with `q * XB` below about -8 contributes the constant `log(eps) = -36.04` instead of `log(Phi(z))`, which keeps falling (-53.2 at z = -10, -804.6 at z = -40). The score uses the same clipped `cdf` in the inverse Mills ratio `pdf / cdf`, and `pdf` underflows to 0 first, so the score of those observations is ~0 where the true value grows like `|z|`. `hessian` and `hessian_factor` do not clip at all, so `pdf / cdf` becomes `0 / 0 = nan` below about z = -38.

Practical consequences with one extreme, misclassified observation in the sample:

- reported `llf` (and therefore AIC/BIC) is wrong away from the optimum: -155.9 where the exact value is -1924.9 in the example below,
- the score has the wrong sign on the slope, so gradient-based optimizers can stop at spurious points: `fit(method="bfgs")` stops at params `[-0.036, 1.72]` with `llf = -155.1` and `bse = [nan, nan]`, while the exact optimum is `[-0.129, 0.077]` with `llf = -202.7`,
- the Hessian is nan at any params where some `q * XB < -38`.

`Logit` had the analogous overflow fixed via `logaddexp` (gh-3923, and the recent `_log_1pexp` follow-up); the clipping in `Probit` was noted in that thread as something that "could be put into the cdf method directly" but was never replaced. scipy provides the exact, stable pieces: `log_ndtr` for `log(Phi(z))` and `exp(logpdf - log_ndtr)` for the Mills ratio, and `log_ndtr` also accepts complex input so complex-step differentiation keeps working.

#### Code Sample, a copy-pastable example

```python
import numpy as np
from scipy import stats
import statsmodels.api as sm

rng = np.random.default_rng(1)
n = 300
x = rng.standard_normal(n)
y = (1.5 * x + rng.standard_normal(n) > 0).astype(int)
x = np.append(x, 40.0)   # one extreme, misclassified observation
y = np.append(y, 0)
X = np.column_stack([np.ones(n + 1), x])

model = sm.Probit(y, X)
params = np.array([0.0, 1.5])
q = 2 * y - 1
z = q * (X @ params)

print(model.loglike(params), stats.norm.logcdf(z).sum())
# -155.934  vs exact  -1924.904
print(model.score(params))
# [-4.874  7.393]  vs exact  [-64.891  -2393.273]
print(np.isnan(model.hessian(params)).any())
# True
res = model.fit(method="bfgs", disp=0)
print(res.params, res.bse, res.llf)
# [-0.036  1.72]  [nan nan]  -155.067   (exact optimum: [-0.129  0.077], llf -202.674)
```

#### Expected Output

`loglike`, `score` and `hessian` equal the exact quantities (`log_ndtr` based), and `fit` converges to the same optimum as a direct minimization of the exact log-likelihood with finite standard errors, as `Logit` does on the same data.

#### Actual Output

See the inline comments in the code sample: `loglike` -155.934 vs exact -1924.904, score `[-4.874, 7.393]` vs exact `[-64.891, -2393.273]`, nan Hessian, and `fit(method="bfgs")` returning params `[-0.036, 1.72]`, `bse = [nan, nan]`, `llf = -155.067` where the exact optimum is `[-0.129, 0.077]` with `llf = -202.674`.

#### Versions

statsmodels 0.15.0 and current main (2fedaa94), numpy 2.x, scipy 1.x, Python 3.12, macOS arm64.

I have a fix ready that replaces the clipped `cdf` in `loglike`, `loglikeobs`, `score`, `score_obs`, `score_factor`, `hessian`, `hessian_factor` and `_deriv_score_obs_dendog` with `log_ndtr` based computations; PR to follow.

#### Checklist

- [x] I have searched the [issue tracker](https://github.com/statsmodels/statsmodels/issues) for a similar issue and did not find one.
- [x] I have confirmed this bug exists on the [latest released version](https://pypi.org/project/statsmodels/) of statsmodels, or on the `main` branch.

#### AI Disclosure

AI tools were used (Claude / Claude Code) to locate this bug by probing `Probit` against scipy references, to produce the repro, and to draft this report. The numbers above were run and checked by me against `scipy.stats.norm.logcdf` and a direct `scipy.optimize.minimize` of the exact log-likelihood.

### PR DESCRIPTION
**Pull request by haeganm**
- [x] closes #10228
- [x] tests added / passed.
- [x] code/documentation is well formatted.
- [x] properly formatted commit message. See
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message).

#### AI Disclosure

Contributions must comply with the statsmodels [AI Policy](https://www.statsmodels.org/devel/dev/ai-policy.html).

- [ ] No AI tools were used to develop this pull request.
- [x] AI tools were used.
      Tool(s): Claude (Claude Code).
      Used for: locating the bug (probing Probit against scipy references after the Logit gh-3923 fix), drafting the implementation and the two regression tests, running the verification (scipy references, complex-step and finite-difference derivative checks, min-version scipy 1.8.1 check), and drafting this description and the linked issue.
      I have personally read, understood, and can explain every line of this diff, and
      have verified the statistical/numerical correctness of the change.

#### Summary

`Probit` evaluated its log-likelihood as `log(clip(cdf(z), FLOAT_EPS, 1))`, which is flat at `log(eps) = -36.04` for `q * XB` below about -8 while the true `log(Phi(z))` keeps falling (-53 at z = -10, -805 at z = -40). The score used the same clipped `cdf` in the Mills ratio `pdf / cdf`, where `pdf` underflows first, so extreme observations contributed ~0 score instead of a value growing like `|z|`; `hessian` and `hessian_factor` did not clip at all and returned nan below z = -38. With a single extreme misclassified observation in the sample this gave a wrong `llf`, a wrong-signed score, a nan Hessian, and `fit(method="bfgs")` stopping at a spurious optimum with nan `bse` (details and repro in #10228). This is the Probit counterpart of the `Logit` overflow fixed in gh-3923.

The fix computes everything in log space with the exact scipy pieces:

- `_norm_logcdf(z) = log_ndtr(z)` replaces `log(clip(cdf))` in `loglike` and `loglikeobs`.
- `_norm_mills(z) = exp(norm._logpdf(z) - log_ndtr(z))` replaces the clipped `pdf / cdf` in `score`, `score_obs`, `score_factor`, `hessian` and `hessian_factor`, so the inverse Mills ratio stays finite and accurate where `cdf` underflows (it behaves like `-z` there).
- `_deriv_score_obs_dendog` uses `exp(logpdf - log_ndtr(z) - log_ndtr(-z))` for `pdf / (cdf * (1 - cdf))`, via `1 - Phi(z) = Phi(-z)`.

`log_ndtr` accepts complex input (checked on scipy 1.8.1, the minimum supported version), so complex-step differentiation of the new `loglike` still works and agrees with the analytic score. In the normal range the new values agree with the old ones to ~1e-14, and all existing Probit tests pass unchanged.

#### Tests

`test_probit_loglikeobs_extreme_linpred` mirrors the gh-3923 Logit test (z from -1e4 to 1e4: finite `loglikeobs`, `score_obs`, `hessian`; agreement with `scipy.stats.norm.logcdf` at 1e-14 and with the Mills ratio reference at 1e-12). `test_probit_extreme_observation_fit` uses the outlier sample: exact `loglike`/`score`/`hessian` against scipy references, complex-step and finite-difference derivative checks, and both `newton` and `bfgs` converging to the same optimum as a direct `scipy.optimize.minimize` of the exact log-likelihood with finite `bse`. Both tests fail on main and pass with the fix. The reference values are scipy's `log_ndtr`/`norm.logcdf` and `norm.logpdf`, which are the same quantities R's `pnorm(log.p = TRUE)` and `dnorm(log = TRUE)` compute.

`pytest statsmodels/discrete/tests` and `statsmodels/tools/tests/test_numdiff.py` pass; `ruff check` is clean. No release note added since the recent BUG merges did not touch `docs/source/release/`; happy to add one wherever you want it.

### CODE REVIEW

**Comment by bashtage** [comment]
LGTM.  Just want to see the other CI tools pass.
